src/league_gui.cpp

```diff
@@ -0,0 +1,451 @@
+		this->InitNested(table);
+	}
+
+	void SetStringParameters(int widget) const override
+	{
+		if (widget != WID_SLT_CAPTION) return;
+		SetDParamStr(0, this->title);
+	}
+
+	void OnPaint() override
+	{
+		this->DrawWidgets();
+	}
+
+	void DrawWidget(const Rect &r, int widget) const override
+	{
+		if (widget != WID_SLT_BACKGROUND) return;
+
+		auto lt = LeagueTable::GetIfValid(this->table);
+		if (lt == nullptr) return;
+
+		Rect ir = r.Shrink(WidgetDimensions::scaled.framerect);
+
+		if (!lt->header.empty()) {
+			SetDParamStr(0, lt->header);
+			ir.top = DrawStringMultiLine(ir.left, ir.right, ir.top, UINT16_MAX, STR_JUST_RAW_STRING, TC_BLACK);
+		}
+
+		int icon_y_offset = (this->line_height - this->icon_size.height) / 2;
+		int text_y_offset = (this->line_height - FONT_HEIGHT_NORMAL) / 2;
+
+		uint rank_left, rank_right, icon_left, text_left, text_right, score_left, score_right;
+		int spacer = WidgetDimensions::scaled.framerect.left + WidgetDimensions::scaled.framerect.right;
+		if (_current_text_dir == TD_RTL) {
+			rank_right = ir.right - WidgetDimensions::scaled.framerect.left;
+			rank_left = rank_right - this->rank_width;
+			icon_left = rank_left - this->icon_size.width;
+			text_right = icon_left - spacer;
+			text_left = text_right - this->text_width;
+			score_left = ir.left + WidgetDimensions::scaled.framerect.right;
+			score_right = score_left + this->score_width;
+		} else {
+			rank_left = ir.left + WidgetDimensions::scaled.framerect.left;
+			rank_right = rank_left + this->rank_width;
+			icon_left = rank_right + spacer;
+			text_left = icon_left + this->icon_size.width;
+			text_right = text_left + this->text_width;
+			score_right = ir.right - WidgetDimensions::scaled.framerect.right;
+			score_left = score_right - this->score_width;
+		}
+
+		for (auto [rank, lte] : this->rows) {
+			DrawString(rank_left, rank_right, ir.top + text_y_offset, rank + STR_ORDINAL_NUMBER_1ST, rank == 0 ? TC_WHITE : TC_YELLOW);
+			if (this->icon_size.width > 0 && lte->company != INVALID_COMPANY) DrawCompanyIcon(lte->company, icon_left, ir.top + icon_y_offset);
+			SetDParamStr(0, lte->text);
+			DrawString(text_left, text_right, ir.top + text_y_offset, STR_JUST_RAW_STRING, TC_BLACK);
+			SetDParamStr(0, lte->score);
+			DrawString(score_left, score_right, ir.top + text_y_offset, STR_JUST_RAW_STRING, TC_BLACK, SA_RIGHT);
+			ir.top += this->line_height;
+		}
+
+		if (!lt->footer.empty()) {
+			SetDParamStr(0, lt->footer);
+			DrawStringMultiLine(ir.left, ir.right, ir.top, UINT16_MAX, STR_JUST_RAW_STRING, TC_BLACK);
+		}
+	}
+
+	void UpdateWidgetSize(int widget, Dimension *size, const Dimension &padding, Dimension *fill, Dimension *resize) override
+	{
+		if (widget != WID_SLT_BACKGROUND) return;
+
+		auto lt = LeagueTable::GetIfValid(this->table);
+		if (lt == nullptr) return;
+
+		this->icon_size = GetSpriteSize(SPR_COMPANY_ICON);
+		this->line_height = std::max<int>(this->icon_size.height + ScaleGUITrad(2), FONT_HEIGHT_NORMAL);
```

Comment:
Could be `WidgetDimensions::scaled.fullbevel.Vertical()` if you want to avoid a magic 2.

(`fullbevel` is `bevel` but always scaled, regardless of the scaled bevels option...)